File: exercism/javascript/anagram/anagram.js
export const findAnagrams = (givenWord, words) => {
	const parsedWords = parseAllWords(givenWord, words);

	let anagrams = parsedWords.filter((word) => {
		const sortedGivenWord = Array.from(givenWord.toLowerCase()).sort().join('');
		let sortedWord = Array.from(word.toLowerCase()).sort().join('');
		if (sortedWord === sortedGivenWord) {
			return word;
		}
	});
	return anagrams;
};

function parseAllWords(givenWord, words) {
	return words.join(' ').split(/[\s+]/).filter((word) => word.toLowerCase() != givenWord.toLowerCase());
}

File: exercism/python/anagram/anagram.py
from collections import Counter


def find_anagrams(word, candidates):
    sorted_given_word = "".join([letter for letter in sorted(word.lower())])
    anagrams = {}
    for candidate in candidates:
        if candidate.lower() == word.lower():
            continue
        anagram = is_anagram(candidate, sorted_given_word)
        if anagram and candidate.lower() not in anagrams:
            anagrams[candidate.lower()] = candidate
    return [word for word in anagrams.values()]


def is_anagram(candidate, sorted_given_word):

    sorted_candidate = "".join([letter for letter in sorted(candidate.lower())])

    return Counter(sorted_given_word) == Counter(sorted_candidate)
